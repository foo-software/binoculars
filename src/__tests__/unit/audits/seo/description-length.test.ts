import ArtifactsInterface from '../../../../interfaces/Artifacts';
import DescriptionLengthAudit from '../../../../audits/seo/description-length';

describe('DescriptionLengthAudit', () => {
  describe('DescriptionLengthAudit.meta', () => {
    it('should match snapshot', () => {
      expect(DescriptionLengthAudit.meta).toMatchSnapshot();
    });
  });

  describe('DescriptionLengthAudit.audit', () => {
    let artifacts: ArtifactsInterface = {};

    beforeAll(() => {
      artifacts = {};
    });

    it('should return a passing score when description is appropriate length', () => {
      artifacts.MetaElements = [
        {
          name: 'description',
          content:
            'ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem',
        },
      ];
      const result = DescriptionLengthAudit.audit(artifacts);
      expect(result.score).toEqual(1);
    });

    it('should return a failing score when description is too long', () => {
      artifacts.MetaElements = [
        {
          name: 'description',
          content:
            'ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum',
        },
      ];
      const result = DescriptionLengthAudit.audit(artifacts);
      expect(result.score).toEqual(0);
    });

    it('should return a failing score when description is too short', () => {
      artifacts.MetaElements = [
        {
          name: 'description',
          content: 'ipsum lorem',
        },
      ];
      const result = DescriptionLengthAudit.audit(artifacts);
      expect(result.score).toEqual(0);
    });

    it('should return a failing score when description is empty', () => {
      artifacts.MetaElements = [
        {
          name: 'description',
          content: '',
        },
      ];
      const result = DescriptionLengthAudit.audit(artifacts);
      expect(result.score).toEqual(0);
    });

    it('should return a failing score when description non-existent', () => {
      artifacts.MetaElements = [];
      const result = DescriptionLengthAudit.audit(artifacts);
      expect(result.score).toEqual(0);
    });
  });
});
