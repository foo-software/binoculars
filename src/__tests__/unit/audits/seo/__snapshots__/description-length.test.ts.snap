// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DescriptionLengthAudit DescriptionLengthAudit.meta should match snapshot 1`] = `
Object {
  "description": "A description provides an extended summary of a page and should have the right amount of content for search engines to parse. Description should be between 100 - 160 characters",
  "failureTitle": "Description is not between 100 - 160 characters",
  "id": "description-length",
  "requiredArtifacts": Array [
    "MetaElements",
  ],
  "title": "Description length is appropriate",
}
`;
